replica/replica.go

```diff
@@ -962,6 +969,23 @@ func (r *Replica) createDisk(name string, userCreated bool, created string) erro
 	r.volume.UserCreatedSnap = append(r.volume.UserCreatedSnap, userCreated)
 	r.activeDiskData = append(r.activeDiskData, &newHeadDisk)
 
+	// assign it to temp variable, so that	r.info remains
+	// intact in case of crash
+	info := r.info
+	info.Head = newHeadDisk.Name
+	info.Dirty = true
+	info.Parent = newSnapName
+	info.RevisionCounter = newHeadDisk.RevisionCounter
+
+	// new head encoded to metadata
+	if err := r.encodeToFile(&info, volumeMetaData); err != nil {
+		return err
+	}
+
+	// crash at this point leads to stale old head and it meta file
```

Comment:
add a TODO over here
// Crash here, stale old head and corresponding metafile will remain,
// which will not be part of the chain. This needs to be removed on next
// restart